inst/shiny/modules/tab_visuals.R

```diff
@@ -435,45 +425,61 @@ tab_visuals_server <- function(id, data, grouping_vars, res_nca) {
 
     # Reactive expression for summary table based on selected group and parameters
     summary_stats <- reactive({
-      req(input$summary_groupby, input$select_display_parameters)
+      req(input$summary_groupby)
       req(res_nca())
 
+      classification_cols <- sort(c(grouping_vars(), res_nca()$data$dose$columns$dose))
+      classification_cols <- classification_cols[
+        classification_cols %in% names(res_nca()$data$conc$data)
+      ]
+      # Join subject data to allow the user to group by it
+      cols_to_join <- c(classification_cols, unname(unlist(res_nca()$data$conc$columns$groups)))
+      stats_data <- inner_join(
+        res_nca()$result,
+        select(res_nca()$data$conc$data, any_of(cols_to_join))
+      ) %>%
+        filter(!(type_interval == "manual" & PPTESTCD != "aucint.last")) %>%
+        mutate(PPTESTCD = case_when(
+          type_interval == "manual" ~ paste0(PPTESTCD, signif(start), "-", signif(end)),
+          TRUE ~ PPTESTCD
+        ))
+
       # Calculate summary stats and filter by selected parameters
-      calculate_summary_stats(res_nca(), input$summary_groupby) %>%
-        filter(PPTESTCD %in% input$select_display_parameters)  %>%
-        rename(PARAM = PPTESTCD)
+      calculate_summary_stats(stats_data, input$summary_groupby)
     })
 
+    observeEvent(summary_stats(), {
+      req(summary_stats())
+      # Update the select display parameters picker input
+      updatePickerInput(
+        session,
+        "select_display_parameters",
+        choices = setdiff(colnames(summary_stats()), "Statistic"),
+        selected = setdiff(colnames(summary_stats()), "Statistic")
```

Comment:
issue: groupby columns are already in another input selected, should not be in select_display_parameters

```suggestion
        choices = setdiff(colnames(summary_stats()), c("Statistic", input$summary_groupby)),
        selected = setdiff(colnames(summary_stats()), c("Statistic", input$summary_groupby))
```